testing/citest/tests/kube_smoke_test.py

```diff
@@ -390,6 +390,190 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
     result.update(kwargs)
     return result
 
+  def create_statefulset_pipeline(self):
+      bindings = self.bindings
+
+      payload = self.agent.make_json_payload_from_kwargs(
+          job=[{
+              'type': 'manual',
+              'user': '[anonymous]'
+          }],
+          description='Create StatefulSet'.format(
+              self.__lb_name,
+              bindings['SPINNAKER_KUBERNETES_ACCOUNT']),
+          application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      (builder.new_clause_builder('StatefulSet Added', retryable_for_secs=60)
+       .get_resources(
+          'statefulsets', extra_args=[self.TEST_APP, '--namespace', self.TEST_NAMESPACE])
+       .contains_path_value('kind', "StatefulSet"))
+      return st.OperationContract(
+          self.new_post_operation(
+              title='create_statefulset', data=payload, path='pipelines/' + self.TEST_APP + '/statefulset-pipeline',
+              status_class=st.SynchronousHttpOperationStatus),
+          contract=builder.build())
+
+  def save_statefulset_pipeline(self):
+      deploy_stage = self.make_deploy_stage_statefulset()
+      job = dict(
+          keepWaitingPipelines='false',
+          application=self.TEST_APP,
+          name='statefulset-pipeline',
+          lastModifiedBy='anonymous',
+          limitConcurrent='true',
+          parallel='true',
+          stages=[deploy_stage]
+      )
+      payload = self.agent.make_json_payload_from_kwargs(**job)
+      expect_match = {key: jp.EQUIVALENT(value)
+                      for key, value in job.items()}
+      expect_match['stages'] = jp.LIST_MATCHES(
+          [
+              jp.DICT_MATCHES({key: jp.EQUIVALENT(value)
+                               for key, value in deploy_stage.items()})])
+
+      builder = st.HttpContractBuilder(self.agent)
+      (builder.new_clause_builder('Has Pipeline',
+                                  retryable_for_secs=15)
```

Comment:
I think 15 is good here, though you have 60 above where I think they underlying asynchronous stability issues (I assume eventual consistency in front50) are the same. You have 15 below as well so perhaps the original should be 15.